master/buildbot/test/unit/test_util.py

```diff
@@ -202,8 +202,12 @@ def test_deep(self):
         self.assertEqual(util.flatten([[1, 2], 3, [[4]]]),
                          [1, 2, 3, 4])
 
+    def test_deeply_nested(self):
+        self.assertEqual(util.flatten([5, [6, (7, 8)]]),
+                         [5, 6, 7, 8])
+
     def test_tuples(self):
-        self.assertEqual(util.flatten([(1, 2), 3]), [(1, 2), 3])
+        self.assertEqual(util.flatten([(1, 2), 3]), [1, 2, 3])
 
```

Comment:
I would recommend also adding:
  flatten(tuple)
  flatten(dict)
  flatten(string)

It's good to test each type at the top-level (not inside any other type) as well as inside other types, just to make sure we get the type preservation at each level. Also, adding strings will make sure that any changes to the implementation in the future dont accidentally listify strings ("abc"->['a','b','c']).